tests/test_transforms.py

```diff
@@ -258,3 +261,19 @@ def transform_class(cls):
                 import UserDict
         """
         )
+
+    def test_transform_aborted_if_recursion_limited(self):
+        def transform_call(node: Call) -> Const:
+            return node
+
+        self.transformer.register_transform(
+            nodes.Call, transform_call, _looks_like_dataclass_field_call
+        )
+        original_limit = sys.getrecursionlimit()
+        sys.setrecursionlimit(1000)
```

Comment:
Notice that pytest does use a higher limit, as @DanielNoord pointed out before. But I'd rather delegate this to users at first; we can always tinker later.

Reply:
pylance doesn't seem to use a higher limit.